pkg/config/expand.go

```diff
@@ -510,39 +411,29 @@ func validateModuleSettingReference(bp Blueprint, mod Module, r Reference) error
 		return nil
 	}
 	g := bp.ModuleGroupOrDie(mod.ID)
-	callingModuleGroupIndex := slices.IndexFunc(bp.DeploymentGroups, func(d DeploymentGroup) bool { return d.Name == g.Name })
 
-	targetModuleGroupIndex, err := modToGrp(bp.DeploymentGroups, r.Module)
+	tm, err := bp.Module(r.Module)
 	if err != nil {
 		return err
 	}
-	targetModuleGroup := bp.DeploymentGroups[targetModuleGroupIndex]
+	tg := bp.ModuleGroupOrDie(tm.ID)
 
 	// references must refer to the same or an earlier group;
-	if targetModuleGroupIndex > callingModuleGroupIndex {
-		return fmt.Errorf("%s: %s is in the later group %s", errorMessages["intergroupOrder"], r.Module, targetModuleGroup.Name)
+	if bp.groupOrder(tg) > bp.groupOrder(g) {
+		return fmt.Errorf("%s: %s is in the later group %s", errorMessages["intergroupOrder"], r.Module, tg.Name)
 	}
 
 	// at this point, we have a valid intragroup or intergroup references to a
```

Comment:
This comment now belongs to the code block below. The Packer error is self-explanatory and does not need its own comment. In principle, the Packer error is a subset of the "output not found" error, but I like making it explicit.

Reply:
Removed all comments